docs/chymyst01.md

```diff
@@ -386,43 +395,164 @@ Sometimes it appears that repeating input molecules is the most natural way of e
 However, I believe it is always possible to introduce some new auxiliary molecules and to rewrite the “chemistry laws” so that input molecules are not repeated while the resulting computations give the same results.
 This limitation could be lifted in a later version of `JoinRun` if it proves useful to do so.
 
-## Order of reactions
+## Order of reactions and nondeterminism
 
-When there are several different reactions that can start the available molecules, the runtime engine will choose the reaction at random.
-In the current implementation of `JoinRun`, the runtime will choose reactions at random, so that every reaction has an equal chance of starting.
+When there are several different reactions that can start the available molecules, the runtime engine will choose the reaction at random,
+so that every reaction has an equal chance of starting.
 
 Similarly, when there are several copies of the same molecule that can be consumed as input by a reaction, the runtime engine will make a choice of which copy 
 of the molecule to consume.
-Currently, `JoinRun` will _not_ randomize the input molecules but make an implementation-dependent choice.
+Currently, `JoinRun` will _not_ fully randomize the input molecules but make an implementation-dependent choice.
 A truly random selection of input molecules may be implemented in the future.
 
-It is not possible to assign priorities to reactions or to molecules.
+Importantly, it is _not possible_ to assign priorities to reactions or to molecules.
 The order of reactions in a join definition is ignored, and the order of molecules in the input list is also ignored.
-The debugging facility will print the molecule names in alphabetical order, and reactions will be printed in an unspecified order.
+Just for the purposes of debugging, molecules will be printed in alphabetical order of names, and reactions will be printed in an unspecified order.
 
-The result of this is that the order in which reactions will start is non-deterministic and unknown.
+The result is that the order in which reactions will start is non-deterministic and unknown.
 
-If the priority of certain reactions is important for a particular application, it is the programmer's task to design the “chemical laws” in such a way that those reactions start in the desired order.
+If the priority of certain reactions is important for a particular application, it is the programmer's task to design the chemical laws in such a way that those reactions start in the desired order.
 This is always possible by using auxiliary molecules and/or guard conditions.
 
+In fact, a facility for assigning priority to molecules or reactions would be self-defeating.
```

Comment:
Do you mean that the following discussion needs to be first introduced in some way? Or is it good as it is?